eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/core/timer/HashedWheelTimer.java

```diff
@@ -0,0 +1,802 @@
+ * limitations under the License.
+ */
+
+package org.apache.eventmesh.runtime.core.timer;
+
+import static org.apache.eventmesh.common.Constants.OS_NAME_KEY;
+import static org.apache.eventmesh.common.Constants.OS_WIN_PREFIX;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link Timer} optimized for approximated I/O timeout scheduling.
+ *
+ * <h3>Tick Duration</h3>
+ * <p>
+ * As described with 'approximated', this timer does not execute the scheduled
+ * {@link TimerTask} on time.  {@link HashedWheelTimer}, on every tick, will
+ * check if there are any {@link TimerTask}s behind the schedule and execute
+ * them.
+ * <p>
+ * You can increase or decrease the accuracy of the execution timing by
+ * specifying smaller or larger tick duration in the constructor.  In most
+ * network applications, I/O timeout does not need to be accurate.  Therefore,
+ * the default tick duration is 100 milliseconds, and you will not need to try
+ * different configurations in most cases.
+ *
+ * <h3>Ticks per Wheel (Wheel Size)</h3>
+ * <p>
+ * {@link HashedWheelTimer} maintains a data structure called 'wheel'.
+ * To put simply, a wheel is a hash table of {@link TimerTask}s whose hash
+ * function is 'deadline of the task'.  The default number of ticks per wheel
+ * (i.e. the size of the wheel) is 512.  You could specify a larger value
+ * if you are going to schedule a lot of timeouts.
+ *
+ * <h3>Do not create many instances.</h3>
+ * <p>
+ * {@link HashedWheelTimer} creates a new thread whenever it is instantiated and
+ * started.  Therefore, you should make sure to create only one instance and
+ * share it across your application.  One of the common mistakes, that makes
+ * your application unresponsive, is to create a new instance for every connection.
+ *
+ * <h3>Implementation Details</h3>
+ * <p>
+ * {@link HashedWheelTimer} is based on
+ * <a href="http://cseweb.ucsd.edu/users/varghese/">George Varghese</a> and
+ * Tony Lauck's paper,
+ * <a href="http://cseweb.ucsd.edu/users/varghese/PAPERS/twheel.ps.Z">'Hashed
+ * and Hierarchical Timing Wheels: data structures to efficiently implement a
+ * timer facility'</a>.  More comprehensive slides are located
+ * <a href="http://www.cse.wustl.edu/~cdgill/courses/cs6874/TimingWheels.ppt">here</a>.
+ */
+public class HashedWheelTimer implements Timer {
```

Comment:
1. Added the execution time callback when executing the Timeout task. The reason for this change is that the existing EventMesh will print the execution time of the retry task, which is compatible through this callback.

2. Since the original logic used the JDK finalize method, which was deprecated in JDK9 and above, this callback was removed and integrated with the existing `AbstractRetryer` lifecycle.

---

1. 新增执行Timeout任务时的执行时间回调, 这个改动的原因是现有的EventMesh会打印重试任务的执行时间, 通过此回调进行了兼容.

2. 由于原先的逻辑使用了JDK的finalize方法, 而在JDK9及以上是废弃的函数, 因此去除了此回调, 通过与现有的`AbstractRetryer`的生命周期进行了集成.
另外, 虽然目前只是改动了上述两个点, 后续可能还需要做一些扩展, 因此隔离开是比较好的选择.

Reply:
Sorry, wrong before. it's actually refer to `org.apache.eventmesh.runtime.core.timer.TimerTask#setExecuteTimeHook`

Reply:
> > Added the execution time callback when executing the Timeout task.
> 
> Does this refer to `Timer#pendingTimeouts()`?

Sorry, wrong before. it's actually refer to `org.apache.eventmesh.runtime.core.timer.TimerTask#setExecuteTimeHook`

Reply:
Please refer to org.apache.eventmesh.runtime.core.timer.HashedWheelTimer, line636
The execution time will be set into the RetryContext when the TimerTask is actually running.

---

在此处会通过Hook方法, 在TimerTask实际运行时将执行时间设置进RetryContext中。